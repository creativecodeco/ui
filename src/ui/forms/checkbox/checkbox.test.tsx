import { render } from '@testing-library/react';
import Checkbox from './checkbox.component';

describe('<Checkbox />', () => {
  it('snapshot', () => {
    const { container } = render(<Checkbox name='test' />);

    expect(container).toMatchSnapshot();
  });

  it('snapshot full props', () => {
    const { container } = render(
      <Checkbox
        name='test'
        checked
        color='primary'
        size='lg'
        isError
        error='Error test'
        position='right'
        disabled
        label='Label'
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('label', () => {
    const label = 'Hello Word';
    const { getByText } = render(<Checkbox name='test' label={label} />);

    expect(getByText(label)).toBeInTheDocument();
  });

  it('with error', () => {
    const error = 'Error message';
    const { getByText } = render(
      <Checkbox name='test' isError error={error} />
    );

    expect(getByText(error)).toBeInTheDocument();
  });

  it('disabled', () => {
    const { getByTestId } = render(<Checkbox name='test' disabled />);

    expect(getByTestId('test')).toHaveProperty('disabled', true);
  });

  it('position right', () => {
    const label = 'Hello Word';
    const { getByText } = render(
      <Checkbox name='test' label={label} position='right' />
    );

    expect(getByText(label)).toBeInTheDocument();
  });

  it('color', () => {
    const { container } = render(<Checkbox name='test' color='primary' />);

    expect(container.querySelector('input')?.className).toEqual(
      'checkbox checkbox-color-primary checkbox-size-md'
    );
  });

  it('size', () => {
    const { container } = render(<Checkbox name='test' size='lg' />);

    expect(container.querySelector('input')?.className).toEqual(
      'checkbox checkbox-size-lg'
    );
  });
});
