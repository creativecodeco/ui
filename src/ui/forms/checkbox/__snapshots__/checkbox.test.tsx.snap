// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> snapshot 1`] = `
<div>
  <label
    class="label cursor-pointer"
    for="test"
  >
    <input
      class="checkbox checkbox-size-md"
      data-testid="test"
      id="test"
      name="test"
      type="checkbox"
    />
  </label>
</div>
`;

exports[`<Checkbox /> snapshot full props 1`] = `
<div>
  <label
    class="label cursor-not-allowed"
    for="test"
  >
    <span
      class="label-text"
    >
      Label
    </span>
    <input
      checked=""
      class="checkbox checkbox-color-primary checkbox-size-lg"
      data-testid="test"
      disabled=""
      id="test"
      name="test"
      type="checkbox"
    />
    <p
      class="text-red-500"
    >
      Error test
    </p>
  </label>
</div>
`;
